mesonbuild/backend/ninjabackend.py

```diff
@@ -1994,8 +1994,8 @@ def generate_rust_target(self, target: build.BuildTarget) -> None:
         # Rustc replaces - with _. spaces or dots are not allowed, so we replace them with underscores
         args += ['--crate-name', target.name.replace('-', '_').replace(' ', '_').replace('.', '_')]
         depfile = os.path.join(self.get_target_private_dir(target), target.name + '.d')
-        args += ['--emit', f'dep-info={depfile}']
-        args += ['--emit', f'link'={target_name}']
+        args += rustc.get_dependency_gen_args(target_name, depfile)
+        args += rustc.get_output_args(target_name)
```

Comment:
... did we just never use get_output_args for rust before? Wow...

Reply:
Because rust outputs a `link=` directive, every other use of `get_output_args` is format agnostic.

Reply:
In addition to what Dylan said, Rust doesn't use many Compiler methods because it has only one compiler. This series introduces some because rustdoc is a "second compiler".